src/couch_replicator/src/couch_replicator_api_wrap.erl

```diff
@@ -68,20 +68,25 @@ db_uri(Db) ->
 
 
 db_open(Db, Options) ->
-    db_open(Db, Options, false).
+    db_open(Db, Options, false, []).
 
-db_open(#httpdb{} = Db1, _Options, Create) ->
+db_open(#httpdb{} = Db1, _Options, Create, CreateParams) ->
     {ok, Db} = couch_replicator_httpc:setup(Db1),
     try
         case Create of
         false ->
             ok;
         true ->
-            send_req(Db, [{method, put}],
+            Db2 = if CreateParams == [] -> Db; true ->
```

Comment:
Let's pull this into a separate function and handle the CreateParams=[] in the head.

`Db2 = maybe_append_create_query_params(Db, CreateParams)` ?